packages/gatsby/src/utils/babel-loader-helpers.ts

```diff
@@ -98,17 +122,34 @@ const prepareOptions = (babel, options = {}, resolve = require.resolve) => {
   ]
 }
 
-const mergeConfigItemOptions = ({ items, itemToMerge, type, babel }) => {
+type mergeConfigItemOptionsType = ({
```

Comment:
As above, rather than typing the whole function like this, type the props object.